### PR TITLE
fix: Tabs overflow handling

### PR DESCRIPTION
**Pull request by ieedan**
Currently the tabs component uses flex-wrap so the tabs stack on overflow.

- Updated the tabs component to properly handle horizontal scrolling
- Added `no-scrollbar` class to hide the scrollbar
- Added `preBlur` and `postBlur` to show the user that there is more content to be scrolled.
- Added example with overflow (Vercel doesn't have this so I can axe it after review if necessary)

### CODE REVIEW

**Review by shyakadavis — COMMENTED** [review]
Hey, @ieedan 

Always appreciate your help. 🙂 Thank you very much for thoughtful polishes. Left 2 suggestions, please let me know what you think.

Also, liked your shadcn-svelte phone-input. If you'd like, please consider adding it to the list here: https://github.com/shyakadavis/awesome-shadcn-svelte 🙂

File: src/lib/components/ui/tabs/tabs-list.svelte [hunk 1]
```diff
@@ -9,16 +10,54 @@
...
+	};
+
+	onMount(() => {
+		scroll(); // initially determine blur state
+		el.addEventListener('scroll', scroll); // on:scroll listener doesn't work on TabsPrimitive
+
+		return () => {
+			el.removeEventListener('scroll', scroll);
+		};
+	});
```
[the remark this answers is not in the document]
Comment (ieedan) [reply src/lib/components/ui/tabs/tabs-list.svelte:41]: Yeah I'll try it out. One thing I forgot to ass is resize handling so I'll add that as well.

File: src/lib/components/ui/tabs/tabs-list.svelte [hunk 2]
```diff
@@ -9,16 +10,54 @@
...
 	{...$$restProps}
 >
+	{#if preBlur}
+		<span class="absolute left-4 h-10 w-8 bg-gray-100 bg-opacity-80 blur-lg"></span>
+	{/if}
 	<slot />
+	<!-- This indicates to the user that there is more content -->
+	{#if postBlur}
+		<span class="absolute right-4 h-10 w-8 bg-gray-100 bg-opacity-80 blur-lg"></span>
+	{/if}
```
[the remark this answers is not in the document]
Comment (ieedan) [reply src/lib/components/ui/tabs/tabs-list.svelte:62]: Yeah it is pretty subtle I'll check them out

**Comment by ieedan** [comment]
Thanks I should have some updates in an hour or two once I get back from lunch

File: src/lib/components/ui/tabs/tabs-list.svelte (hunk 2 above)
[the remark this answers is not in the document]
Comment (shyakadavis) [reply src/lib/components/ui/tabs/tabs-list.svelte:62]: Coincidentally, here are more that I literally just landed on now 😅 (Not progressive, but still)

- https://x.com/moritzpetersen/status/1823046608390590595
- https://codepen.io/moritzpetersen/pen/qBzVrBv

[the remark this answers is not in the document]
Comment (ieedan) [reply src/lib/components/ui/tabs/tabs-list.svelte:59]: image

Well I know the last one won't work lmao.

File: src/lib/components/ui/tabs/tabs-list.svelte (hunk 1 above)
[the remark this answers is not in the document]
Comment (ieedan) [reply src/lib/components/ui/tabs/tabs-list.svelte]: So actually on testing the scroll does not capture elements I think its page scroll

**Comment by ieedan** [comment]
I tried a few of the solutions you sent I couldn't find anything much better than what I already have. I darkened up the shadow though so now it is more visible.

+ Added transitions to in and out of blurs
+ Resize handler
+ `blur_overflow` Prop to allow for turning off the blurring

To make it work I have to wrap it in a `<div class="relative">` which isn't very nice.

Let me know what you think!

**Comment by shyakadavis** [comment]
Hey, @ieedan 

You're right. No easy solution would have worked without adding an unnecessary `div`, so instead of foregoing on the `indicate oveerflow` bit, let's use a `mask` instead. I know adding a package for just a mask is overkill, but I suspect this will come in handy in other places/future, and for what it's worth, the plugin is only a few lines-ish.

**Comment by ieedan** [comment]
Yeah that looks much better

File: src/lib/components/ui/tabs/tabs-list.svelte [hunk 3]
```diff
@@ -1,24 +1,66 @@
...
-<!-- TODO: Add a no-scrollbar class. -->
+<svelte:window on:resize={handle_mask} />
+
+<!-- Make sure you have the `no-scrollbar` class in your tailwind.config file -->
 <TabsPrimitive.List
 	data-variant={variant}
+	bind:el
 	class={cn(
-		'group inline-flex w-full flex-wrap items-baseline gap-3 overflow-hidden overflow-x-auto p-1 text-gray-900 data-[variant=default]:[box-shadow:_0_-1px_0_var(--accents-2)_inset;]',
+		'group flex w-full items-baseline gap-3 overflow-hidden overflow-x-auto p-1 text-gray-900 no-scrollbar data-[variant=default]:[box-shadow:_0_-1px_0_var(--accents-2)_inset;]',
```
Comment (ieedan) [inline src/lib/components/ui/tabs/tabs-list.svelte:54]: yeah this is just a better approach overall. I was thinking about it all wrong

**Review by shyakadavis — APPROVED** [review]
As always, grateful for the help, @ieedan 

Thank you.

**Comment by ieedan** [comment]
image

I think we need 2 masks for situations like this

**Comment by shyakadavis** [comment]
Ah, I see. Sorry for missing this.

Mind opening a follow-up P.R please? I was working on the `Table` component, and might get to it late.

**Comment by ieedan** [comment]
Yeah np.

I think this is too much? 

image

**Comment by shyakadavis** [comment]
Might be indeed. 😅 Even if we were to reduce the `mask-point-to-[{percetage}]` it could still be too much. 🤷

**Comment by ieedan** [comment]
I am gonna play with it a bit more hopefully I can get it to match up with what we have already

**Comment by ieedan** [comment]
My best idea at this point is to use a Pseudo-element to do this. However I am having a lot of trouble doing this right now and even if I can get the classes to apply we would still need to apply it to a parent container of the tabs because it moves during scroll.

I think I'll have to circle back at another time.

**Pull request merged**